app/controllers/image-set.js

```diff
@@ -0,0 +1,73 @@
+import Ember from 'ember';
+import {imageTypes, genders} from 'lion-guardians/utils/units';
+
+export default Ember.Controller.extend({
+  imageTypes: imageTypes,
+  genders: genders,
+  uploadImageType: imageTypes[0],
+  uploadIsPublic: false,
+  isEditing: false,
+
+  organizations: function() {
+    return this.store.findAll('organization');
+  }.property(),
+
+  actions: {
+    addImage: function(upload){
+      var url = upload.get('url'),
+          isPublic = this.get('uploadIsPublic'),
+          imageType = this.get('uploadImageType'),
+          model = this.get('model');
+
+      model.addImage(url, isPublic, imageType);
+
+      Ember.run.later(this, function(){
+        // delete upload so it disappears from upload window
+        upload.deleteRecord();
+      }, 3000);
```

Comment:
thanks for the clarifying comment — why the delay, though? If we do need a delay, the value should be in the `config/environment.js` instead of hardcoded here.